spotify-conversion-next/pages/_app.tsx

```diff
@@ -1,12 +1,25 @@
-import { SiteLayout } from "@/components";
 import "@/styles/globals.css";
-import { ChakraProvider } from "@chakra-ui/react";
+import { Box, ChakraProvider, Container } from "@chakra-ui/react";
 import type { AppProps } from "next/app";
+import { SessionProvider } from "next-auth/react";
+import NavigationBar from "@/components/nav/NavigationBar";
 
-export default function App({ Component, pageProps }: AppProps) {
+export default function App({
+  Component,
+  pageProps: { session, pageProps },
+}: AppProps) {
   return (
     <ChakraProvider>
-      <Component {...pageProps} />
+      <SessionProvider session={session}>
```

Comment:
Think of the `<SessionProvider>` as something that wraps our whole application. Everything wrapped inside of it will have access to the `session` variable through the `useSession()` react hook (look how it's used on next-auth's  [front page](https://next-auth.js.org/). If `session` is null, then you aren't logged in.

You can see me conditionally rendering things in the views based on whether session is null or not.